### PR TITLE
Unblock JDK-next (JDK18) acceptance build

### PR DESCRIPTION
**Pull request by babsingh**
**1. Signature of NativeLibraries.load has changed in JDK18** 

Update the signature of NativeLibraries.load in native code for JDK18.
 
**2. Set the useDirectMethodHandle Java flag to false** 

Setting jdk.reflect.useDirectMethodHandle to false disables the new JEP 416
changes where the reflection API is dependent on the MethodHandle API.

This flag will be reenabled once OpenJ9 supports JEP 416.

Related: #13852
Related: #13880
 
**3. Update MHN.getMemberVMInfo to work with MemberName.vminfoIsConsistent** 

MemberName.vminfoIsConsistent is only used in an assertion to validate the
vmtarget and vmindex elements of MemberName. It relies on MHN.getMemberVMInfo
to retrieve the vmtarget and vmindex elements of MemberName.

For correct validation, MHN.getMemberVMInfo should return 0 for vmindex when
the vTableIndex is 0 and ref kind is invokevirtual or invokeinterface.
Otherwise, it should return -1 for vmindex. MHN.getMemberVMInfo has been
updated to perform this behaviour. This in return fixes the behaviour of
MemberName.vminfoIsConsistent.

Closes: #13877
 
**4. InjectedInvoker behaviour modified in JDK18** 

In JDK17, InjectedInvoker does not have the NESTMATE attribute. In JDK18,
InjectedInvoker's definition has been updated, and it is given the NESTMATE
attribute. The new changes invalidate the filter that is used to identify
InjectedInvoker classes for modifying its class name. The filter has been
updated to work with JDK18.

Closes: #13878

Co-authored-by: Jason Feng <fengj@ca.ibm.com>
Co-authored-by: Jack Lu <Jack.S.Lu@ibm.com>
Signed-off-by: Babneet Singh <sbabneet@ca.ibm.com>

### CODE REVIEW

**Comment by babsingh** [comment]
Started [Personal build #11090](https://hyc-runtimes-jenkins.swg-devops.com/view/OpenJ9%20-%20Personal/job/Pipeline-Build-Test-Personal/11090/) to validate that the above changes will unblock the JDK-next acceptance build.

fyi @JasonFengJ9 @fengxue-IS @tajila @pshipton @keithc-ca

**Comment by pshipton** [comment]
Started a non-promoting acceptance build with these changes https://openj9-jenkins.osuosl.org/job/Pipeline-OpenJDK-Acceptance/147/

**Comment by pshipton** [comment]
jenkins test sanity zlinux jdk11,jdk17

**Comment by pshipton** [comment]
jenkins compile win32 jdk8

File: runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp
Comment (JasonFengJ9) [inline runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp:1508]: Please consider put the rvalue on the left.

File: runtime/compiler/env/j9method.cpp
Comment (JasonFengJ9) [inline runtime/compiler/env/j9method.cpp:3007]: We need a JIT reviewer here.

fyi @0xdaryl

File: jcl/src/java.base/share/classes/java/lang/System.java
Comment (keithc-ca) [inline jcl/src/java.base/share/classes/java/lang/System.java:606]: Is this intended to be temporary? If so, a comment would be welcome.

File: runtime/bcutil/ROMClassBuilder.cpp
Comment (keithc-ca) [inline runtime/bcutil/ROMClassBuilder.cpp:298]: Please add a comment explaining why the condition needs to be inverted beginning with Java 18.

**Comment by pshipton** [comment]
All the testing I started looks good. I don't expect we'll need to re-run it all if the only changes are comments and formatting.

**Review by 0xdaryl — APPROVED** [review]
I approve the JIT-specific code.

File: jcl/src/java.base/share/classes/java/lang/System.java
Comment (babsingh) [reply jcl/src/java.base/share/classes/java/lang/System.java:611]: Temporary. A comment has been added.

File: runtime/bcutil/ROMClassBuilder.cpp
Comment (babsingh) [reply runtime/bcutil/ROMClassBuilder.cpp:304]: A comment has been added.

**Review by babsingh — COMMENTED** [review]
@JasonFengJ9 @keithc-ca All feedback has been addressed.

File: runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp
Comment (fengxue-IS) [inline runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp:1508]: Maybe add a comment for why virtual/interface have different vmindex value

**Comment by babsingh** [comment]
### Build links

1. [Pull Request - OpenJ9](https://openj9-jenkins.osuosl.org/job/PullRequest-OpenJ9/1141/)
2. [Test_openjdk17_j9_sanity.functional_s390x_linux_Personal](https://openj9-jenkins.osuosl.org/job/Test_openjdk17_j9_sanity.functional_s390x_linux_Personal/44/)
3. [Build_JDK11_s390x_linux_Personal](https://openj9-jenkins.osuosl.org/job/Build_JDK11_s390x_linux_Personal/249/)
4. [Build_JDK17_s390x_linux_Personal](https://openj9-jenkins.osuosl.org/job/Build_JDK17_s390x_linux_Personal/86/)
5. [Build_JDK8_x86-32_windows_Personal](https://openj9-jenkins.osuosl.org/job/Build_JDK8_x86-32_windows_Personal/116/)
6. [Test_openjdk11_j9_sanity.functional_s390x_linux_Personal](https://openj9-jenkins.osuosl.org/job/Test_openjdk11_j9_sanity.functional_s390x_linux_Personal/124/)

File: runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp
Comment (babsingh) [reply runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp:1516]: Comments added.

**Review by fengxue-IS — APPROVED** [review]

**Review by JasonFengJ9 — APPROVED** [review]
LGTM
Thanks @babsingh

**Comment by keithc-ca** [comment]
I think the test failure in https://openj9-jenkins.osuosl.org/job/Pipeline_Build_Test_JDK17_s390x_linux/200/ is eclipse-openj9/openj9#11293.

**Review by keithc-ca — APPROVED** [review]

**Comment by pshipton** [comment]
@tajila once you approve, pls leave the merging to me. I think we'll need to run the acceptance build again as we've already got new content, and jdknext needs to promote at the same time as this is merged or nightly builds will be broken.

**Review by tajila — APPROVED** [review]

**Pull request merged**
